src/groups/group_action.jl

```diff
@@ -10,67 +10,135 @@ abstract type AbstractGroupAction{AD<:ActionDirection} end
 
 The group that acts in action `A`.
 """
-function base_group(A::AbstractGroupAction)
-    error("base_group not implemented for $(typeof(A)).")
-end
-
-function similar_result(A::AbstractGroupAction, f, x...)
-    return similar_result(base_group(A), f, x...)
-end
+base_group(A::AbstractGroupAction) = error("base_group not implemented for $(typeof(A)).")
 
 """
     g_manifold(A::AbstractGroupAction)
 
 The manifold the action `A` acts upon.
 """
-function g_manifold(A::AbstractGroupAction)
-    error("g_manifold not implemented for $(typeof(A)).")
-end
+g_manifold(A::AbstractGroupAction) = error("g_manifold not implemented for $(typeof(A)).")
+
+similar_result(A::AbstractGroupAction, f, x...) = similar_result(g_manifold(A), f, x...)
 
 """
-    apply(A::AbstractGroupAction, x, a)
+    direction(::AbstractGroupAction{AD}) -> AD
+
+Get the direction of the action
+"""
+direction(::AbstractGroupAction{AD}) where {AD} = AD()
+
+"""
+    apply!(A::AbstractGroupAction, y, a, x)
 
 Apply action `a` to the point `x` with the rule specified by `A`.
 The result is saved in `y`.
 """
-function apply!(A::AbstractGroupAction, y, x, a)
+function apply!(A::AbstractGroupAction{LeftAction}, y, a, x)
     error("apply! not implemented for action $(typeof(A)) and points $(typeof(y)), $(typeof(x)) and $(typeof(a)).")
 end
+function apply!(A::AbstractGroupAction{RightAction}, y, a, x)
+    ainv = inv(base_group(A), a)
+    return apply!(switch_direction(A), y, ainv, x)
+end
 
-"""
-    apply(A::AbstractGroupAction, x, a)
+@doc doc"""
+    apply(A::AbstractGroupAction, a, x)
 
 Apply action `a` to the point `x`. The action is specified by `A`.
+Unless otherwise specified, right actions are defined in terms of the left action. For
+point $x ∈ M$ and action element $a$, the right action is
+
+````math
+x ⋅ a ≐ a^{-1} ⋅ x.
+````
 """
-function apply(A::AbstractGroupAction, x, a)
+function apply(A::AbstractGroupAction, a, x)
     y = similar_result(A, apply, x, a)
-    apply!(A, y, x)
-    return y
+    return apply!(A, y, a, x)
 end
 
-function compose(A::AbstractGroupAction{LeftAction}, a, b)
-    return compose(base_group(A), a, b)
+"""
```

Comment:
For readability one could switch the words `inverse` and `apply`, also for the diff cases later. Just as an idea.

Reply:
I like this idea. Also because it makes like functions near one another in alphabetical order. We would also need to change `inverse_translate => translate_inverse` and `inverse_translate_diff => translate_diff_inverse` (or `translate_inverse_diff`). @mateuszbaran what do you think?

Reply:
I don't have a preferred word order here, feel free to choose whichever looks the best for you.

Reply:
Actually, maybe this isn't best. e.g. we already have `inverse_retract` so until now, `inverse_` has been used as a prefix not a suffix.

Reply:
As I think for `exp` of the exponential_map for me `retract` is more like `retraction`(`_map`) so then the `inverse` is an adjective to that so in front while for  the `apply` it's more an adverb, so there it's after the verb, but yes it's inconsistent in the sense that `retract` in its form as written is also a verb of course.

Reply:
I don't follow the argument. We have two maps, a retraction and an action. The retraction map is not explicitly represented as an object, but its action can be called with the verb `retract`. The action on a manifold is also a map; it is, however, explicitly represented, so we have this helper verb function `apply` to actually call its action. I don't see a major difference between these two functions.

Reply:
And that's ok, I see that my feeling is not consistent and it stays a feeling. I liked the inverted version of `apply_inverse(::Action)`, but that's merely a feeling and not even an oppinion.